### PR TITLE
10792 rsection stirrups bars

### PR DESCRIPTION
**Pull request by david-liska**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [ ] Attached examples
- [ ] Own example attached to Pull Request

**Test Configuration**:
* RFEM / RSTAB / RSECTION version:
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: includes/BasicObjects/Bar.js [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+ * @param {Number} material_no      Material number
+ * @param {Number} layer_no         Layer number
+ * @param {String} comment          Comment, can be undefined
+ * @param {Object} params           Parameters, can be undefined
+ */
+function Bar (no,
+    start_point_no,
+    end_point_no,
+    material_no,
+    layer_no,
```
Comment (jarabroz) [inline includes/BasicObjects/Bar.js:24]: Add Layer object. User needs to create layer first. Then we can add bars to layer.

File: includes/BasicObjects/Bar.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    return this.bar;
+};
+
+/**
+ * Creates Multi uniform Bar
+ * @param {Number} no                               Number of Bar, can be undefined
+ * @param {Number} start_point_no                   Start point number
+ * @param {Number} end_point_no                     End point number
+ * @param {Number} material_no                      Material's number
+ * @param {Number} layer_no                         Number of layer
```
Comment (jarabroz) [inline includes/BasicObjects/Bar.js:52]: In constructor

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js [hunk 3]
```diff
@@ -116,6 +115,30 @@ stressPoint.OnLine(undefined, 6, [0.75, undefined]);
 stressPoint.OnLine(undefined, 6, [undefined, 0.01, false]);     // absolute distances
 stressPoint.OnElement(undefined, 5, [0.10, undefined]);   // relative distances
 
+/*********************************************** Bar ********************************************/
```
Comment (jarabroz) [inline examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:118]: make separete demo for concrete stuff

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js [hunk 4]
```diff
@@ -116,6 +115,30 @@ stressPoint.OnLine(undefined, 6, [0.75, undefined]);
...
+var bar5 = new Bar();
+bar5.SinglePoint(undefined, 40, material2.GetNo(), 1, undefined, 0.05, 0.05);  // With default diameter
+
+/***************************************** Bar + Stirrups****************************************/
+var concreteMaterial = new Material(undefined, "C12/15");
+var reinforcedSteelMaterial = new Material(undefined, "Grade 40");
+rsPart.Rectangle(undefined, [0.3, -0.18], 0.28, 0.11, concreteMaterial.GetNo());
+var stirrup = new Stirrup(undefined, [501, 502, 503, 504, 501], reinforcedSteelMaterial.GetNo(), 0.022, 0.022, 1);
+var bar6 = new Bar();
+bar6.SinglePoint(undefined, 511, material2.GetNo(), 1);  // With default diameter and offsets
```
Comment (jarabroz) [inline examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:137]: Here I prefer to use also bar6 then bar7 etc. Then it allows users to use GetNo and GetBar

**Review by jarabroz — CHANGES_REQUESTED** [review]

**Review by jarabroz — CHANGES_REQUESTED** [review]
Make separate demo

File: includes/BasicObjects/Bar.js [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+ * @constructor
+ * @param {Number} no               Number of Bar, can be undefined
+ * @param {Number} start_point_no   Start point number
+ * @param {Number} end_point_no     End point number
+ * @param {Number} material_no      Material number
+ * @param {Number} layer_no         Layer number
+ * @param {String} comment          Comment, can be undefined
+ * @param {Object} params           Parameters, can be undefined
+ */
+function Bar (no,
```
[the remark this answers is not in the document]
Comment (david-liska) [reply includes/BasicObjects/Bar.js:20]: There are new functions StartPoint and EndPoint created.

File: includes/BasicObjects/Bar.js [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+Bar.prototype.GetBar = function () {
+    return this.bar;
+};
+
+/**
+ * Creates Multi uniform Bar
+ * @param {Number} no                               Number of Bar, can be undefined
+ * @param {Number} start_point_no                   Start point number
+ * @param {Number} end_point_no                     End point number
+ * @param {Number} material_no                      Material's number
```
[the remark this answers is not in the document]
Comment (david-liska) [reply includes/BasicObjects/Bar.js:51]: It means the constructor should be created always with paramaters? Shared parameter: material_no + layer_no, comment, params?

File: includes/BasicObjects/Bar.js (hunk 2 above)
Comment (david-liska) [reply includes/BasicObjects/Bar.js:52]: Need to disscuss.

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js (hunk 3 above)
Comment (david-liska) [reply examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:119]: Created (concrete_reinforcement_demo.js)

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js (hunk 4 above)
Comment (david-liska) [reply examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:138]: Bars are moved to new js demo file, created new bar's objects.

**Review by jarabroz — APPROVED** [review]
Ready to merge

**Pull request merged**